src/storage/src/render/persist_sink.rs

```diff
@@ -1219,6 +1220,7 @@ where
                 source_statistics
                     .inc_updates_committed_by(batch_metrics.inserts + batch_metrics.retractions);
                 source_statistics.update_snapshot_committed(&batch_upper);
+                source_statistics.update_rehydration_latency_ms(&batch_upper);
```

Comment:
does this occur even if there is no new data after rehydration? I believe an empty batch is appended, right? It might be nice to instead call this on `Progress` updates?

Reply:
Yep! `compare_and_append_batch` is how the shards upper is advanced, so even if we don't have data for a given time, we still need to append an empty batch to make sure the frontier keeps advancing and readers are notified about progress.

I prefer keeping the statistics update here, for two reason:
* This is also where we update `snapshot_committed`. Both statistics are similar, they basically only differ in which frontier we compare the current one to. So it makes sense to treat them similarly too.
* Here we know for a fact that the frontier has durably advanced. When we handle the `Progress` update above, it could theoretically happen that we get stuck/delayed between there and the compare-and-append and then we might report hydration even though it hasn't been completed. Admittedly, the chance of this happening is low.

Reply:
sounds good!